src/main/java/org/javarosa/xform/parse/StandardBindAttributesProcessor.java

```diff
@@ -124,25 +123,25 @@ DataBinding createBinding(IXFormParserFunctions parserFunctions, FormDef formDef
         return binding;
     }
 
-    private Condition buildCondition(String xpath, String type, IDataReference contextRef) {
+    private Triggerable buildCondition(String xpath, String type, IDataReference contextRef) {
         final int trueAction;
         final int falseAction;
         final String prettyType;
 
         switch (type) {
             case "relevant":
-                prettyType  = "display";
-                trueAction  = Condition.ACTION_SHOW;
+                prettyType = "display";
+                trueAction = Condition.ACTION_SHOW;
                 falseAction = Condition.ACTION_HIDE;
                 break;
             case "required":
```

Comment:
When looking at cycle-checking behavior, we've considered relevants and calculates but actually requireds and readonlys could also lead to cycles, right? I guess there's no particular need to test for those since they all behave the same.

Reply:
I'll throw extra test cases for those too. I prefer to be comprehensive. Otherwise, the lack of tests might be misguiding or confusing for future readers.